### DISCUSSION TITLE
Backward pass of the Sliding Window Inferer (gradient calculation)

### DISCUSSION THREAD

**Discussion by matt3o** [Project-MONAI/MONAI#7059] [General]
I was wondering with my tutor @Zrrr1997, how exactly the Sliding Window Inferer backwards pass works. As far as we know, the operation should not be differentiable. However it does work.
Even more, in our case, using an overlap of 0.25 increased the Dice by 1% compared to an overlap of 0 on the same validation overlap of 0.25.
Also, how does that work for the overlapped regions and how are the gradients calculated there?

**Comment by wyli** [comment]
these are implemented with Pytorch APIs and we try to keep the gradient info as much as possible e.g. https://github.com/Project-MONAI/MONAI/pull/6034, the overlapped regions are (weighted) averages of different model forwards so the gradients are linearly combined with respect to the model.

**Reply by matt3o** [reply]
Thanks!
